src/values/FunctionValue.js

```diff
@@ -13,25 +13,17 @@ import type { ObjectKind } from "../types.js";
 import type { LexicalEnvironment } from "../environment.js";
 import type { Realm } from "../realm.js";
 import { ObjectValue, NumberValue } from "./index.js";
-import * as t from "babel-types";
-import type { BabelNodeLVal, BabelNodeBlockStatement, BabelNodeSourceLocation } from "babel-types";
 import invariant from "../invariant.js";
 
+/* Abstract base class for all function objects */
 export default class FunctionValue extends ObjectValue {
   constructor(realm: Realm, intrinsicName?: string) {
     super(realm, realm.intrinsics.FunctionPrototype, intrinsicName);
   }
 
   $ConstructorKind: void | string;
```

Comment:
A big part of this whole refactoring is to separate out the aspects of a function-given-as-source into a separate subclass to avoid numerous defined-ness checks and instead get checked type-safety.